rhp/v4/server.go

```diff
@@ -708,61 +707,49 @@ func (s *Server) handleRPCRenewContract(stream net.Conn) error {
 		renterInputSum = renterInputSum.Add(si.SiacoinOutput.Value)
 	}
 
-	// validate the renter added enough inputs
-	if !renterInputSum.Equals(renterCost) {
+	if n := renterInputSum.Cmp(renterCost); n < 0 {
 		return errorBadRequest("expected renter to fund %v, got %v", renterInputSum, renterCost)
+	} else if n > 0 {
+		// if the renter added too much, add a change output
+		renewalTxn.SiacoinOutputs = append(renewalTxn.SiacoinOutputs, types.SiacoinOutput{
+			Address: renewal.NewContract.RenterOutput.Address,
+			Value:   renterInputSum.Sub(renterCost),
+		})
 	}
 
-	fceBasis, fce, err := s.contractor.ContractElement(req.Renewal.ContractID)
+	elementBasis, fce, err := s.contractor.ContractElement(req.Renewal.ContractID)
 	if err != nil {
 		return fmt.Errorf("failed to get contract element: %w", err)
 	}
-	renewalTxn.FileContractResolutions[0].Parent = fce
```

Comment:
We no longer set the element as parent on the resolution in the renewal txn here, while `UpdateV2TransactionSet` calls `updateTxnProofs` which includes a step where update the proof of the resolutions `updateProof(&txn.FileContractResolutions`.. is that an oversight or is that intentional?

Reply:
Oh I guess `UpdateStateElement` covers that

Reply:
That's intentional as we don't want the chain manager to try and update the zero value proof.